src/Symfony/Component/HttpClient/Response/AmpResponse.php

```diff
@@ -171,18 +173,21 @@ private static function perform(ClientState $multi, array &$responses = null): v
      */
     private static function select(ClientState $multi, float $timeout): int
     {
-        $selected = 1;
-        $delay = Loop::delay(1000 * $timeout, static function () use (&$selected) {
-            $selected = 0;
-            Loop::stop();
-        });
-        Loop::run();
+        $start = microtime(true);
+        $remaining = $timeout;
 
-        if ($selected) {
-            Loop::cancel($delay);
-        }
+        while (true) {
+            self::$delay = Loop::delay(1000 * $remaining, [Loop::class, 'stop']);
```

Comment:
@kelunik there is something I don't get here that you might want to check once this PR is merged. It looks like the delay defined here is not respected.
To see what I mean, checkout this PR, then add `dump([$timeout, $remaining]);` after the second "if" below, and run tests with `./phpunit src/Symfony/Component/HttpClient/ --filter Amp.*NotATimeout`.
You'll see something like:
```
array:2 [
  0 => 0.9
  1 => 0.87336587905884
]
```

which means that the delay expired `0.87336587905884` earlier than expected.
Do I miss something or could this be a bug in your event loop?

Reply:
I've added two failing tests in https://github.com/amphp/amp/pull/319, one of those is expected to fail, the other isn't.